### PR TITLE
[14.0][MIG] product_cost_price_avco_sync: Migration to 14.0

### PR DESCRIPTION
**Pull request by zamberjo**
ping @carlosdauden @sergio-teruel @CarlosRoca13

### CODE REVIEW

File: product_cost_price_avco_sync/tests/test_product_cost_price_avco_sync.py [hunk 1]
```diff
@@ -11,66 +11,65 @@
...
-        cls.customer_location = cls.env.ref("stock.stock_location_customers")
-        cls.warehouse = cls.env.ref("stock.warehouse0")
-        cls.stock_location = cls.env.ref("stock.stock_location_stock")
-        cls.picking_type_in = cls.env.ref("stock.picking_type_in")
-        cls.picking_type_out = cls.env.ref("stock.picking_type_out")
-        cls.partner = cls.env["res.partner"].create({"name": "Test Partner"})
-        cls.categ_all = cls.env.ref("product.product_category_all")
-        cls.categ_all.property_cost_method = "average"
-        cls.product = cls.env["product.product"].create(
+    def setUp(self):
```
Comment (chienandalu) [inline product_cost_price_avco_sync/tests/test_product_cost_price_avco_sync.py]: Why the change?

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: product_cost_price_avco_sync/tests/test_product_cost_price_avco_sync.py (hunk 1 above)
Comment (zamberjo) [reply product_cost_price_avco_sync/tests/test_product_cost_price_avco_sync.py]: Sorry, I didn't look at the guide, and I thought the change from `SavepointCase` to `TransactionCase` (which has to be done in the migration to 15.0) was in this version. I fix it. 

I have also fixed a warning I was giving, that a `with_company` was missing.

**Review by carlosdauden — APPROVED** [review]
Ok, thanks

**Review by jessicaatic — APPROVED** [review]
It is functional, I tested it and it is correct

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I have done some changes in 13.0 that would be good to have it here, and the refactoring is still on its way. I will write when the 13.0 version is ready for rebasing this on top of it.

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Pull request merged**